src/interpreter/mod.rs

```diff
@@ -0,0 +1,1320 @@
+use std::collections::hash_map::{Entry, HashMap};
+use std::error;
+use std::fmt;
+
+use crate::convert::cast_usize;
+
+pub use self::func::{Func, FuncFlags, ParamInfo};
+pub use self::value::{Ty, Value};
+
+pub mod ast;
+pub mod func;
+pub mod value;
+
+/// A name resolution error.
+#[derive(Debug, PartialEq)]
+pub enum ResolveError {}
```

Comment:
Yes! (Although depending on our milestones and deadlines it may not happen very soon). 

The gist will be the same as for the runtime error: The most complete information possible will be given so that the UI can pin-point and display the error message on a concrete widget.

I already have a WIP version of the name resolving and its corresponding errors. Some errors will move from the `RuntimeError` to the `ResolveError`, and will be impossible to encounter during program execution (modulo bugs). Similar should later happen for `TypecheckError`.